src/overlays/ovl_DF4260/fox_co.c

```diff
@@ -0,0 +1,3333 @@
+    player->wings.unk_0C = 0.0f;
+    player->wings.unk_04 = 0.0f;
+    Math_SmoothStepToF(&player->unk_130, 0.0f, 0.1f, 15.0f, 0.0f);
+    Math_SmoothStepToF(&player->unk_12C, 0.0f, 0.1f, 15.0f, 0.0f);
+    Math_SmoothStepToF(&player->unk_034, 0.0f, 0.1f, 3.0f, 0.0f);
+    Math_SmoothStepToAngle(&player->unk_4D8, 0.0f, 0.1f, 20.0f, 0.0f);
+    Math_SmoothStepToF(&player->unk_110, 0.0f, 0.1f, 3.0f, 0.0f);
+    if (player->unk_1D0 >= 3) {
+        player->camEye.y += 3.0f;
+        player->camAt.y += 3.0f;
+        player->pos.y += 3.0f;
+        gActors[0].obj.pos.y += 3.0f;
+        gActors[1].obj.pos.y += 3.0f;
+        gActors[2].obj.pos.y += 3.0f;
+    }
+
+    gBgColor = 0x845;
+
+    switch (player->unk_1D0) {
+        case 0:
+            func_8001A38C(1, &player->unk_460);
+            sp54 = player->camEye.x - D_DF4260_8019B6D8[62];
+            sp4C = player->camEye.z - D_DF4260_8019B6D8[64];
+
+            D_80177A48[0] = Math_RadToDeg(Math_Atan2F(sp54, sp4C));
+            D_80177A48[1] = sqrtf(SQ(sp54) + SQ(sp4C));
+            player->unk_1D0++;
+            D_80177A48[5] = 0.0f;
+            D_80177A48[4] = D_80177A48[5];
+            D_80177A48[2] = D_80177A48[5];
+            /* fallthrough */
+        case 1:
+            Math_SmoothStepToF(&player->unk_0E4, 0.0f, 0.1f, 5.0f, 0.0f);
+            Math_SmoothStepToF(&player->pos.y, 400.0f, 0.05f, 3.0f, 0.0f);
+            Math_SmoothStepToF(&D_80177A48[1], 1300.0f, 0.05f, 1000.0f, 0.0f);
+            if (player->unk_0E8 > 180.0f) {
+                D_80177A48[0] += 0.5f;
+            } else {
+                D_80177A48[0] -= 0.5f;
+            }
+            Matrix_RotateY(gCalcMatrix, D_80177A48[0] * M_DTOR, 0);
+            sp64.x = 0.0f;
+            sp64.y = 0.0f;
+            sp64.z = D_80177A48[1];
+            Matrix_MultVec3f(gCalcMatrix, &sp64, &sp58);
+            Math_SmoothStepToF(&player->camEye.x, D_DF4260_8019B6D8[62] + sp58.x, 0.05f, 500.0f, 0.0f);
+            Math_SmoothStepToF(&player->camEye.y, D_DF4260_8019B6D8[63] + 100.0f, 0.05f, 500.0f, 0.0f);
+            Math_SmoothStepToF(&player->camEye.z, D_DF4260_8019B6D8[64] + sp58.z, 0.05f, 500.0f, 0.0f);
+            Math_SmoothStepToF(&player->camAt.x, D_DF4260_8019B6D8[62], 0.05f, 500.0f, 0.0f);
+            Math_SmoothStepToF(&player->camAt.y, D_DF4260_8019B6D8[63], 0.05f, 500.0f, 0.0f);
+            Math_SmoothStepToF(&player->camAt.z, D_DF4260_8019B6D8[64], 0.05f, 500.0f, 0.0f);
+            temp_fa0 = player->pos.x - D_DF4260_8019B6D8[62];
+            temp_fa1 = player->pos.z - D_DF4260_8019B6D8[64];
+            if (D_80177A80 < 30) {
+                var_fv1 = Math_SmoothStepToAngle(&player->unk_0E8, Math_RadToDeg(-Math_Atan2F(temp_fa0, temp_fa1)),
+                                                 0.5f, 4.0f, 0.0001f) *
+                          20.0f;
+            } else {
+                var_fv1 = Math_SmoothStepToAngle(&player->unk_0E8, Math_RadToDeg(Math_Atan2F(temp_fa0, temp_fa1)), 0.5f,
```

Comment:
could a temp be used here for ` Math_RadToDeg(Math_Atan2F(temp_fa0, temp_fa1))` to clean this up a bit?